scripts/core/helpers/xhr.ts

```diff
@@ -0,0 +1,107 @@
+import {logger} from "core/services/logger";
+import ng from 'core/services/ng';
+import {appConfig} from "appConfig";
+
+interface IHttpRequestOptions {
+    method: 'GET' | 'POST' | 'PATCH' | 'PUT';
+    url: string; // absolute url
+    payload?: {};
+    headers?: {[key: string]: any};
+    onSuccess(responseText: string): void;
+    onError?(xhr: XMLHttpRequest): void;
+}
+
+interface IHttpRequestReturnValue {
+    abort(): void;
+}
+
+export function httpRequest(options: IHttpRequestOptions): IHttpRequestReturnValue {
+    const {method, url, payload, headers, onSuccess, onError} = options;
+    const xhr = new XMLHttpRequest();
+
+    xhr.open(method, url, true);
```

Comment:
imo using [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) would be better, it's supported in the browsers we care about

Reply:
The interface doesn't expose any details on which API is being used to talk to the network, so changing which API is being used without changing the interface won't change anything.

What benefits do you expect to gain by using `fetch`?

Reply:
it's cleaner api than xhr, uses promises

Reply:
Cleanliness is [relative](https://gomakethings.com/why-i-still-use-xhr-instead-of-the-fetch-api/) and isn't relevant if you don't interact with it directly as is the case here. 

I've added promise-based versions for all helpers.

Reply:
imo having less code is better, so if we don't need promise-less version I don't see any reason to have it, nor to use xhr, why would we do that?
it could be:
```
export function httpRequest(options): Promise {
    const {method, url, payload, headers} = options;

    return fetch(url, {
        method,
        headers,
        body: payload != null ? JSON.stringify(payload): null,
    })
        .then((response) => response.ok ? response : Promise.reject(response))
        .catch((reason) => logger.error(...));

};
```